modules/local/annotation/main.nf

```diff
@@ -0,0 +1,57 @@
+process TANGRAM_ANNOTATION {
+    label "process_gpu"
+
+    conda "${moduleDir}/environment.yml"
+    container "${workflow.containerEngine == 'apptainer' && !task.ext.singularity_pull_docker_container
+        ? 'docker://quentinblampey/sopa:latest-tangram'
+        : 'docker.io/quentinblampey/sopa:latest-tangram'}"
+
+    input:
+    tuple val(meta), path(sdata_path)
+    file sc_reference
+    val cli_arguments
+
+    output:
+    tuple val(meta), path(sdata_path)
+    path "${sdata_path}/tables/table/obs"
+    path "versions.yml"
+
+    script:
+    """
+    sopa annotate tangram ${sdata_path} --sc-reference-path ${sc_reference} ${cli_arguments}
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        sopa: \$(sopa --version)
+        tangram: \$(python -c "import tangram; print(tangram.__version__)" 2> /dev/null)
+    END_VERSIONS
+    """
+}
+
+process FLUO_ANNOTATION {
```

Comment:
please split this up into different submodules. 
also, it would be great to also have them as nf-core modules